pkg/controller/networkpolicyrecommendation/controller.go

```diff
@@ -194,9 +205,14 @@ func (c *NPRecommendationController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
-	go func() {
-		wait.Until(c.resyncNPRecommendation, npRecommendationResyncPeriod, stopCh)
-	}()
+	c.gcQueue.Add(gcKey{
+		removeStaleDbEntries: true,
+		removeStaleSparkApp:  true,
+		addResync:            true,
+	})
+	go wait.Until(c.gcworker, time.Second, stopCh)
```

Comment:
I think we only add gcKey to gcQueue when theia-manager is started? I'm wondering why do we use wait.Util?
Won't the func processNextGcWorkItem get blocked at line #333?

Reply:
As the key will be added back to the queue when there is an error in `handleStaleResources`, I apply the same logic as the other work queues. And as `processNextGcWorkItem` will return false when `handleStaleResources` succeed, the goroutine will finish at that time and won't be blocked. It should be possible to replace this by a simpler while loop, but I'm not sure if it is a better solution.

Reply:
Thanks YunTang, you're right. Using wait.Until makes the goroutine running until the stopCh. Updated!